workspaces/local-cli/src/shared/local-cli-task-runner.ts

```diff
@@ -241,6 +234,41 @@ export class LocalCliTaskRunner implements IOpticTaskRunner {
     developerDebugLogger(`api base url: ${apiBaseUrl}`);
     const cliClient = new Client(apiBaseUrl);
 
+    ////////////////////////////////////////////////////////////////////////////////
+    const spectacle = new LocalCliSpectacle(apiBaseUrl, opticEngine);
+    const idQuery = await spectacle.query<any>({
+      query: `{
+        metadata {
+          id
+        }
+      }`,
+      variables: {},
+    });
+    const specId = idQuery?.data?.metadata?.id ?? 'anon-spec-id';
+
+    const batchCommitQuery = await spectacle.query<any>({
```

Comment:
Might be good to add a comment on why we do this - it seems really weird if we don't have context

Reply:
done